src/main/java/com/icthh/xm/ms/dashboard/service/bulk/impl/AtomicBulkDashboardServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.icthh.xm.ms.dashboard.service.bulk.impl;
+
+import com.icthh.xm.commons.exceptions.BusinessException;
+import com.icthh.xm.ms.dashboard.domain.Dashboard;
+import com.icthh.xm.ms.dashboard.mapper.DashboardMapper;
+import com.icthh.xm.ms.dashboard.repository.DashboardRepository;
+import com.icthh.xm.ms.dashboard.service.bulk.AtomicBulkDashboardService;
+import com.icthh.xm.ms.dashboard.service.dto.DashboardDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Collection;
+
+import static java.util.stream.Collectors.toList;
+
+@Service
+@RequiredArgsConstructor
+public class AtomicBulkDashboardServiceImpl implements AtomicBulkDashboardService {
+
+    private final DashboardMapper dashboardMapper;
+    private final DashboardRepository dashboardRepository;
+
+    @Override
+    public void create(Collection<DashboardDto> dashboardItems) {
+        Collection<Dashboard> dashboardEntities = dashboardItems.stream()
+            .map(dashboardMapper::toEntity)
+            .collect(toList());
+
+        save(dashboardEntities);
+    }
+
+    @Override
+    public void update(Collection<DashboardDto> dashboardItems) {
+        Collection<Dashboard> dashboardEntities = dashboardItems.stream()
+            .map(this::update)
+            .collect(toList());
+
+        save(dashboardEntities);
+    }
+
+    @Override
+    public void delete(Collection<DashboardDto> dashboardItems) {
+        Collection<Dashboard> dashboardEntities = dashboardItems.stream()
+            .map(dashboardMapper::toFullEntity)
+            .collect(toList());
+
+        deleteAll(dashboardEntities);
+    }
+
+    @Transactional
+    void save(Collection<Dashboard> dashboardEntities) {
+        dashboardRepository.saveAll(dashboardEntities);
+    }
+
+    void deleteAll(Collection<Dashboard> dashboardEntities) {
+        dashboardRepository.deleteAll(dashboardEntities);
+    }
+
+    private Dashboard update(DashboardDto dashboardDto) {
+        Dashboard dashboard = dashboardRepository.findById(dashboardDto.getId())
```

Comment:
as this method is used in cyle it will issue N select to the DB. 
consider to collect all IDs to collection and then use `dashboardRepository.findByIdIn(ids)`
it will issue 1 select to DB.

Reply:
or just use `findAllById(Iterable<ID> ids)` from `SimpleJpaRepository` (it does the same for not composited IDs)

Reply:
In your proposal, we will have: N (collect ids) + N^2 (go for each again and find it in the result of dashboardRepository.findByIdIn(ids) ). If you have a more optimized way, please share it

Reply:
@anton-contributor 
seems you did not get the idea of my proposal. I meant next approach:
```
@Override
    public void update(Collection<DashboardDto> dashboardItems) {
        
        Map<Long, DashboardDto> dtoById = dashboardItems.stream()
                                                    .collect(Collectors.toMap(DashboardDto::getId, Function.identity()));
        
        List<Dashboard> result = dashboardRepository.findAllById(dtoById.keySet())
                                                     .stream()
                                                     .map(dashboard -> dashboardMapper.merge(dtoById.get(dashboard.getId()), dashboard))
                                                     .collect(toList());
        save(result);
    }
```